src/pipelinedefinition.cpp

```diff
@@ -68,10 +68,21 @@ Status PipelineDefinition::validate(ModelManager& manager) {
     if (!validationResult.ok()) {
         return validationResult;
     }
-
+    std::unique_lock lock(metadataMtx);
+    validationResult = updateInputsInfo(manager);
+    if (!validationResult.ok()) {
+        return validationResult;
+    }
+    validationResult = updateOutputsInfo(manager);
+    if (!validationResult.ok()) {
+        return validationResult;
+    }
```

Comment:
I'm wondering if the variable name `validationResult` is appropriate here. It suggests that some more validation is going on in updateInputsInfo and updateOutputsInfo as it is followed by the chain of validate...() calls.